EFCore.CheckConstraints/Internal/ValidationCheckConstraintConvention.cs

```diff
@@ -320,7 +324,29 @@ protected virtual void AddListOfValuesConstraint(
     }
 
     protected virtual string GenerateRegexSql(string columnName, [RegexPattern] string regex)
-        => string.Format(
+    {
+        var sqlServerOptionsExtension  = _dbContextOptions.Extensions
```

Comment:
Rather than doing this each time for each regex, maybe do this once at initialization time and cache a simple `_isSqlServerWithNewRegex` or similar?